### PR TITLE
10219 remove reraise related py2 compatability hacks

### PR DESCRIPTION
**Pull request by graingert**
## Scope and purpose

there were two different ways to reraise an error with a traceback - there's now only 1

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10219#
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [ ] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
* [ ] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-trac-id

Author: <github_username>, <github_usernames_if_more_authors>
Reviewer: <github_username>, <github_username_if_more_reviewers>
Fixes: ticket:<trac_ticket_number>, ticket:<another_if_more_in_one_PR>

Long description providing a summary of these changes.
(as long as you wish)
```

### CODE REVIEW

File: src/twisted/enterprise/adbapi.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def _runWithConnection(self, func, *args, **kw):
                 conn.rollback()
             except BaseException:
                 log.err(None, "Rollback failed")
-            compat.reraise(excValue, excTraceback)
+            raise excValue.with_traceback(excTraceback)
```
Comment (graingert) [inline src/twisted/enterprise/adbapi.py]:
```suggestion
            raise
```

File: src/twisted/enterprise/adbapi.py [hunk 2]
```diff
@@ -453,7 +453,7 @@ def _runInteraction(self, interaction, *args, **kw):
...
```
Comment (graingert) [inline src/twisted/enterprise/adbapi.py]:
```suggestion
            raise
```

File: src/twisted/enterprise/adbapi.py (hunk 1 above)
Comment (graingert) [reply src/twisted/enterprise/adbapi.py]: pretty sure 0-arg `raise` takes the `__traceback__` from the correct place here anyway...

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks. I think that you can update to the zero arg variant for `raise`.

**Review by adiroiban — APPROVED** [review]
A minor failure
> src/twisted/enterprise/adbapi.py:10:1: F401 'sys' imported but unused

do you have pre-commit hook enabled ?

I have to see why this is not raised via local the pre-commit hook

Please fix the error and merge.

**Comment by graingert** [comment]
> A minor failure
> 
> > src/twisted/enterprise/adbapi.py:10:1: F401 'sys' imported but unused
> 
> do you have pre-commit hook enabled ?
> 
> I have to see why this is not raised via local the pre-commit hook
> 
> Please fix the error and merge.

I was making the changes via GitHub web

File: src/twisted/enterprise/adbapi.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
 import sys
```
Comment (graingert) [inline src/twisted/enterprise/adbapi.py]:
```suggestion
```

**Comment by adiroiban** [comment]
> I was making the changes via GitHub web

understood... I guess that a pre-commit helper is needed to auto-delete unused imports :)

**Comment by adiroiban** [comment]
Thanks. I think that this is ready for merge :)

**Pull request merged**